GVRf/Framework/src/org/gearvrf/GVRContext.java

```diff
@@ -628,118 +634,144 @@ public AiNode wrapSceneNode(Object parent, Object matrix,
 
         AiNode rootNode = assimpScene.getSceneRoot(wrapperProvider);
 
-        List<AiNode> childrenNodes = rootNode.getChildren();
+        // Recurse through the entire hierarchy to attache all the meshes as
+        // Scene Object
+        this.recurseAssimpNodes(assetRelativeFilename, wholeSceneObject,
+                rootNode, wrapperProvider);
+
+        return wholeSceneObject;
+    }
+
+    /**
+     * Helper method to recurse through all the assimp nodes and get all their
+     * meshes that can be used to create {@link GVRSceneObject} to be attached
+     * to the set of complete scene objects for the assimp model.
+     * 
+     * @param assetRelativeFilename
+     *            A filename, relative to the {@code assets} directory. The file
+     *            can be in a sub-directory of the {@code assets} directory:
+     *            {@code "foo/bar.png"} will open the file
+     *            {@code assets/foo/bar.png}
+     * 
+     * @param wholeSceneObject
+     *            A reference of the {@link GVRSceneObject}, to which all other
+     *            scene objects are attached.
+     * 
+     * @param node
+     *            A reference to the AiNode for which we want to recurse all its
+     *            children and meshes.
+     * 
+     * @param wrapperProvider
+     *            AiWrapperProvider for unwrapping Jassimp properties.
+     * 
+     */
 
+    public void recurseAssimpNodes(
+            String assetRelativeFilename,
```

Comment:
Do we need to set this API public?

Thanks in advance.

Sincerely
-Chao Chen